controller/src/beerocks/master/db/db.cpp

```diff
@@ -2513,6 +2513,17 @@ bool db::add_client_to_persistent_db(const sMacAddr &mac,
                    << " already exists but with different type";
         return false;
     }
+
+    if (m_persistent_db_clients_count >= config.clients_persistent_db_max_size) {
+        LOG(DEBUG) << "reached max clients size in persistent db - removing a client before adding "
+                      "new client";
+        if (!remove_candidate_client()) {
```

Comment:
I suggest something in-between:
```
bool db::set_old_client_selection_func(old_client_selection_func user_selection_func) {
  m_user_selection_func = user_selection_func;
  return true;
}

Class db {
private:
  old_client_selection_func user_selection_func;
}
```

and then the set operation will be performed as part of the init flow of the controller.

Reply:
yes, as I said: _few_ ways. 
* The important issue - let the client supply the functionality of "how" to select the one to be removed.
* What is the type of `old_client_selection_func`?
* I prefer the general `user_selection` and attach it to each function that requires such a function. Because in the future the user might need to provide more functionality to solve more "how to" things for the database. Then the `user_selection` idiom is well understood and you don't need to invent a new member and a new interface each time such a need is introduced. But totally up to you.

Reply:
as discussed offline, I have decided to leave it as is at this stage - but will consider introducing this change if we really require more than one way of determining the candidate client to be removed.